### PR TITLE
feat(mgmt): add restart support

### PR DESCRIPTION
**Pull request by wwhai**
Fixes <issue-number>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt_cli.erl [hunk 1]
```diff
@@ -661,3 +673,29 @@ listener_identifier(Protocol, ListenOn) ->
...
+                restart(emqx_mgmt_http, http, Port, Options);
+            "https:management" ->
+                restart(emqx_mgmt_http, https, Port, Options);
+            _ -> ok
+        end
+    end, application:get_env(emqx_management, listeners, [])).
+
+restart_emqx_dashboard(Identifier) ->
+    lists:foreach(fun({Protocol ,Port, Options}) ->
+        case Identifier = atom_to_list(Protocol) ++ ":dashboard" of
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_cli.erl:690]: this  variable `Identifier` is clashing with the one at line 688.

Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_cli.erl:690]:
```
case Protocol =:= atom_to_list(Protocol) ++ ":dashboard" of
    true ->
        restart(emqx_dashboard, Protocol, Port, Options);
    false ->
        ok
end
```

**Review by zmstone — CHANGES_REQUESTED** [review]

File: apps/emqx_management/src/emqx_mgmt_cli.erl [hunk 2]
```diff
@@ -661,3 +673,23 @@ listener_identifier(Protocol, ListenOn) ->
...
+restart_emqx_management(Identifier) ->
+    lists:foreach(fun({Protocol ,Port, Options}) ->
+        case Identifier =:= atom_to_list(Protocol) ++ ":management" of
+            true -> restart(emqx_management, Protocol, Port, Options);
+            false -> ok
+        end
+    end, application:get_env(emqx_management, listeners, [])).
+
+restart_emqx_dashboard(Identifier) ->
+    lists:foreach(fun({Protocol ,Port, Options}) ->
```
Comment (terry-xiaoyu) [inline apps/emqx_management/src/emqx_mgmt_cli.erl:686]: Reformat the comma and spaces:

```suggestion
    lists:foreach(fun({Protocol, Port, Options}) ->
```

File: apps/emqx_management/src/emqx_mgmt_cli.erl [hunk 3]
```diff
@@ -661,3 +673,23 @@ listener_identifier(Protocol, ListenOn) ->
         ID ->
             ID
     end.
+
+restart_emqx_management(Identifier) ->
```
Comment (terry-xiaoyu) [inline apps/emqx_management/src/emqx_mgmt_cli.erl:677]: I noticed that restart_emqx_management/1 and restart_emqx_dashboard/1 are almost in the same structure. Then rewrite it into a common function would make sense.

**Review by terry-xiaoyu — CHANGES_REQUESTED** [review]

File: apps/emqx_management/src/emqx_mgmt_cli.erl [hunk 4]
```diff
@@ -509,6 +509,18 @@ listeners(["stop", _Proto, ListenOn]) ->
...
+
+listeners(["restart", "https:management" = Identifier]) ->
+    restart_http_listener(Identifier);
+
+listeners(["restart", "http:dashboard" = Identifier]) ->
+    restart_http_listener(Identifier);
+
+listeners(["restart", "https:dashboard" = Identifier]) ->
+    restart_http_listener(Identifier);
+
```
Comment (terry-xiaoyu) [inline apps/emqx_management/src/emqx_mgmt_cli.erl:523]:
```suggestion
listeners(["restart", "http:dashboard" = Identifier]) ->
    restart_http_listener(http, emqx_dashboard);
listeners(["restart", "https:dashboard" = Identifier]) ->
    restart_http_listener(https, emqx_dashboard);
   
```

**Review by terry-xiaoyu — CHANGES_REQUESTED** [review]

**Comment by zmstone** [comment]
commit log looks unrelated.
e.g.
feat(rule-engine): reformat code style

**Review by zmstone — APPROVED** [review]

**Pull request merged**